### PR TITLE
Fixes #26584 - create FormField using pf components

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Adds `FormField` based on patternfly-react components (bootstrap ones).
It should replace current `CommonForm` component as it is not using patternfly, but it is rather custom implementation.
It uses this component  to render `datetime_local_f` from rails form.

This PR should be followed by rewriting all current inputs to it, but first I would stabilize options, their naming and conventions, as I didn't covered all use cases and definitely not in the best way :)
Please consider and comment mainly that while reviewing, thanks! :)

Tests needed (anticipated impacts):
* Report generation - generatate_at
* Adding own form with date/dateTime/time form fields (react and rails one).

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/common/forms/FormField.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      <ControlLabel className={labelSizeClass}>
+        {label}
+        {labelHelp && (
+          <FieldLevelHelp
+            placement="right"
+            buttonClass="field-help"
+            content={<React.Fragment>{labelHelp}</React.Fragment>}
+          />
+        )}
+        {required && '*'}
```
Comment (xprazak2) [inline webpack/assets/javascripts/react_app/components/common/forms/FormField.js]: nit: you may need to swap `labelHelp` and `required`: https://github.com/theforeman/foreman/pull/6495

File: webpack/assets/javascripts/react_app/components/common/forms/FormField.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      </ControlLabel>
+      <Col className={inputSizeClass}>
+        <FormControl
+          componentClass={inputComponents[type]}
```
Comment (xprazak2) [inline webpack/assets/javascripts/react_app/components/common/forms/FormField.js]: Note that this restricts form fields to those that are in core. Plugins will not be able to use their own input components if they need [something special](https://github.com/theforeman/foreman_templates/pull/109/files#diff-5f4a0359daf2426a6b0874b78886c28cR31).

Comment (ezr-ondrej) [reply webpack/assets/javascripts/react_app/components/common/forms/FormField.js]: Well, that is not really true, we would just have to figure a way for them to be able to register new type, what should not be so hard to do.

File: webpack/assets/javascripts/react_app/components/common/forms/FormField.js (hunk 1 above)
Comment (ezr-ondrej) [reply webpack/assets/javascripts/react_app/components/common/forms/FormField.js]: :+1: thanks

File: webpack/assets/javascripts/react_app/components/common/forms/FormField.js (hunk 2 above)
Comment (ezr-ondrej) [reply webpack/assets/javascripts/react_app/components/common/forms/FormField.js]: Ok, I have done both aproaches. You can choose now. Maybe I could add even one - higher order component wrapping any field up nicely?

Comment (xprazak2) [reply webpack/assets/javascripts/react_app/components/common/forms/FormField.js]: Registering would work, but I do not see what benefits it brings us when passing children achieves the same and has no downsides as far as I can see. But maybe that is just a personal preference.

Comment (ezr-ondrej) [reply webpack/assets/javascripts/react_app/components/common/forms/FormField.js]: The advantage is simply in not exposing all of the children in `componentRegistry` and still be able to reach them from Rails. Like this, we have only one mount point :)

**Comment by ezr-ondrej** [comment]
[test katello]

File: app/helpers/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+!tags
```
Comment (xprazak2) [inline app/helpers/.gitignore:1]: Is `/app/helpers/tags` not picked up by git for some reason?

File: webpack/assets/javascripts/react_app/components/common/forms/FormField.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+import { noop } from '../../../common/helpers';
+import DateTimePicker from '../DateTimePicker/DateTimePicker';
+import DatePicker from '../DateTimePicker/DatePicker';
+import TimePicker from '../DateTimePicker/TimePicker';
+
+const inputComponents = {
+  date: DatePicker,
+  dateTime: DateTimePicker,
+  time: TimePicker,
+};
```
Comment (xprazak2) [inline webpack/assets/javascripts/react_app/components/common/forms/FormField.js:22]: Does it make sense to add helpers for the rest of input components?

**Comment by xprazak2** [comment]
Perhaps unrelated, we might need to fix separately: 

date-picker

**Comment by xprazak2** [comment]
Storybook might need some styling:

storybook-field

File: app/helpers/.gitignore (hunk 3 above)
Comment (ezr-ondrej) [reply app/helpers/.gitignore:1]: exactly, there is a global rule for `tags` probably for `ctags` (do not know if it necessary in all levels, not a ctags user :disappointed:

File: webpack/assets/javascripts/react_app/components/common/forms/FormField.js (hunk 4 above)
Comment (ezr-ondrej) [reply webpack/assets/javascripts/react_app/components/common/forms/FormField.js:22]: 1. I wouldn't do it now,
2. It's upon discussion if to support all the inputs from rails, or just those which are not accessible now (rich components) and as you've stated already above from React we can just easily use the `children` prop for other components.

Will do a followup and discuss there, if you agree :slightly_smiling_face:

**Comment by ezr-ondrej** [comment]
> Perhaps unrelated, we might need to fix separately:
> 
> date-picker

Yeah it is a known issue with positioning of datetime picker, but preferrably it should be solved by using https://github.com/patternfly/patternfly-react/pull/1420 instead of our forked version.

**Comment by ezr-ondrej** [comment]
> Storybook might need some styling:

ooops, will do.

**Comment by ezr-ondrej** [comment]
I've fixed the storybook :)

**Comment by xprazak2** [comment]
Storybook looks much better now, I just found one related error in console, maybe it would be worth it to add an empty string into appropriate place to get rid of it.

empty-string

**Comment by ezr-ondrej** [comment]
I made it uncontrolled as a default (`value = undefined`) - what seems reasonable to me, as it is the same behaviour as you would expect in pure React rendered input.

**Review by xprazak2 — DISMISSED** [review]
LGTM

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
merged, díky @ezr-ondrej!

**Pull request merged**